user_sync/rules.py

```diff
@@ -777,38 +870,74 @@ def execute_actions(self):
             if not had_work:
                 break
     
-class Group(object):
-
+class ConfigGroup(object):
     dashboard_groups = {}
 
-    def __init__(self, group_name, organization_name):
+    def __init__(self, group_name, organization_name, designation):
         '''
         :type group_name: str
         :type organization_name: str        
         '''
         self.group_name = group_name
         self.organization_name = organization_name
-        Group.dashboard_groups[(group_name, organization_name)] = self
+        self.designation = designation
+
+        ConfigGroup.dashboard_groups[(group_name, organization_name)] = self
     
+    def key(self):
+        return { 'group_name': self.group_name, 'organization_name': self.organization_name }
```

Comment:
I imagine the performance is negligible mostly... should I use another key format?